bin/p2-pcctl/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/square/p2/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+	"github.com/square/p2/Godeps/_workspace/src/gopkg.in/alecthomas/kingpin.v2"
+	"github.com/square/p2/Godeps/_workspace/src/k8s.io/kubernetes/pkg/labels"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/flags"
+	"github.com/square/p2/pkg/kp/pcstore"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/pc/control"
+	"github.com/square/p2/pkg/pc/fields"
+	"github.com/square/p2/pkg/types"
+)
+
+const (
+	CmdCreate = "create"
+	CmdShow   = "show"
+	CmdDelete = "delete"
+)
+
+var (
+	cmdCreate         = kingpin.Command(CmdCreate, "Create a pod cluster. ")
+	createPodID       = cmdCreate.Flag("pod", "The pod ID on the pod cluster").Required().String()
+	createAZ          = cmdCreate.Flag("az", "The availability zone of the pod cluster").Required().String()
+	createName        = cmdCreate.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+	createSelector    = cmdCreate.Flag("selector", "A pod selector to use for this pod cluster. This will be generated from other arguments if not provided").String()
+	createAnnotations = cmdCreate.Flag("annotations", "Complete set of annotations - must parse as JSON!").String()
+
+	cmdShow      = kingpin.Command(CmdShow, "Show a pod cluster. ")
+	showPodID    = cmdShow.Flag("pod", "The pod ID on the pod cluster").Required().String()
+	showAZ       = cmdShow.Flag("az", "The availability zone of the pod cluster").Required().String()
+	showName     = cmdShow.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+	showSelector = cmdShow.Flag("selector", "A pod selector to use for this pod cluster. This will be generated from other arguments if not provided").String()
```

Comment:
No, it's theoretical at this point. I'm going to remove it until there's a concrete motivation for it.

Aside: is there an convention in open source for user-specified selectors as strings? I'd prefer to re-use a parser to implementing one.